src/service/db/alerts/alert.rs

```diff
@@ -217,19 +229,21 @@ pub async fn watch() -> Result<(), anyhow::Error> {
                 drop(cacher);
             }
             db::Event::Delete(ev) => {
-                let item_key = ev.key.strip_prefix(key).unwrap();
-                let stream_key = item_key[0..item_key.rfind('/').unwrap()].to_string();
-                let alert_name = item_key[item_key.rfind('/').unwrap() + 1..].to_string();
+                let Some((org, stream_type, stream_name, alert_name)) =
+                    cluster::parse_alert_key(&ev.key)
+                else {
+                    log::error!("watch_alerts: failed to parse event key {}", &ev.key);
+                    continue;
+                };
+
                 let mut cacher = STREAM_ALERTS.write().await;
-                if stream_key.contains('/') {
-                    let group = match cacher.get_mut(&stream_key) {
-                        Some(v) => v,
-                        None => continue,
-                    };
-                    group.retain(|v| !v.name.eq(&alert_name));
-                } else {
-                    cacher.remove(&stream_key);
-                }
```

Comment:
It's not obvious to me that this case can ever actually happen, so I do not account for it in the refactored logic. Am I misunderstanding something?

Reply:
i think it's fine since you have the key generation refactored out to follow the same pattern.